### ISSUE TITLE
Binary Deserialization Only Returns 1024 Records of 4895 Records (works correctly using JSON)

### ISSUE THREAD

**Issue by damonwildercarr** [mtth/avsc#188]
When I use AVSC to deserialize avro json I get back all 4895 records. When I try to use the binary encoding, I only get 1024 back. The binary data is in an ArrayBuffer. I use that ArrayBuffer to create a Buffer.

Here is what the code looks like:

  ```
      const objectSChema = avro.Type.forSchema(effectiveSchema,  
            {logicalTypes: {'isodatetime': IsoDateTimeType, 'isodate': IsoDateType, 'decimal': DecimalType }});

        if (result.isBinary()) {

            const buff = Buffer.from(result.payload);     //ArrayBuffer             
            return objectSChema.fromBuffer(buff);       // returns only 1024
            
        }
        else {

            return objectSChema.clone(result.payload);        // returns 4895
        }   
```

I know the endpoint binary is correct as the Java client binary test returns all 4895 records. Someone who knows more about avro then I do said 'perhaps he is not reading past the first block of the array encoding of avro'.

Thanks,
Damon

**Comment by nlongton** [comment]
There is a bug in the ArrayType.prototype._read() method. The inner for loop just keeps writing to the first 1024 entries regardless of the array block. I'm new to github and am trying to push a fix.

**Comment by damonwildercarr** [comment]
Amazing! Thank you! I see you did a pull request - Hopefully it will be released soon as I need this now.

**Comment by mtth** [comment]
Thank you both! I'll take a look later today.

**Comment by mtth** [comment]
Fix released in `5.3.1`. In case you are curious, the issue was with arrays encoded using multiple blocks: entries from subsequent blocks would override previous ones  (`avsc` always uses single blocks when encoding, which is likely why this hasn't come up before).

**Comment by damonwildercarr** [comment]
Awesome - upgrading now :-)


On Jun 22, 2018, at 7:29 PM, Matthieu Monsch <notifications@github.com<mailto:notifications@github.com>> wrote:


Fix released in 5.3.1. In case you are curious, the issue was with arrays encoded using multiple blocks: entries from subsequent blocks would override previous ones (avsc always uses single blocks when encoding, which is likely why this hasn't come up before).

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mtth_avsc_issues_188-23issuecomment-2D399623880&d=DwMCaQ&c=91HTncUBNS9Yv-Uuv2IlCA&r=29Dx056QbZJvQJ7LFM8rlSBLSUyjrMEeXk9-oRceQlU&m=ahyLP-_y_XNk0Tu98XVIF_7HvTAhRXGxJ5YWAS7yFTs&s=by0sV2By7L9T6s1VZxdty4V620bC99lwR51bCRmUlsQ&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AIRVXoF9-5FMQpOkXIRLHijk8d1ulA0ozrks5t-5Faf8gaJpZM4UyzLc&d=DwMCaQ&c=91HTncUBNS9Yv-Uuv2IlCA&r=29Dx056QbZJvQJ7LFM8rlSBLSUyjrMEeXk9-oRceQlU&m=ahyLP-_y_XNk0Tu98XVIF_7HvTAhRXGxJ5YWAS7yFTs&s=GygWMYdOSmfIV2vUchuAZXAK7r02IKJvuPdiGltfmnY&e=>.
This message contains confidential information and is intended only for the individual named. If you are not the named addressee, you should not disseminate, distribute, alter or copy this e-mail. Please notify the sender immediately by e-mail if you have received this e-mail by mistake and delete this e-mail from your system. E-mail transmissions cannot be guaranteed to be secure or without error as information could be intercepted, corrupted, lost, destroyed, arrive late or incomplete, or contain viruses. The sender, therefore, does not accept liability for any errors or omissions in the contents of this message which arise during or as a result of e-mail transmission. If verification is required, please request a hard-copy version. This message is provided for information purposes and should not be construed as a solicitation or offer to buy or sell any securities or related financial instruments in any jurisdiction.  Securities are offered in the U.S. through PIMCO Investments LLC, distributor and a company of PIMCO LLC.

The individual providing the information herein is an employee of Pacific Investment Management Company LLC ("PIMCO"), an SEC-registered investment adviser.  To the extent such individual advises you regarding a PIMCO investment strategy, he or she does so as an associated person of PIMCO.  To the extent that any information is provided to you related to a PIMCO-sponsored investment fund ("PIMCO Fund"), it is being provided to you in the individual's capacity as a registered representative of PIMCO Investments LLC ("PI"), an SEC-registered broker-dealer.  PI is not registered, and does not intend to register, as a municipal advisor and therefore does not provide advice with respect to the investment of the proceeds of municipal securities or municipal escrow investments.  In addition, unless otherwise agreed by PIMCO, this communication and any related attachments are being provided on the express basis that they will not cause PIMCO LLC, or its affiliates, to become an investment advice fiduciary under ERISA or the Internal Revenue Code.
